Runtime Error: Missing CollectionSelector component breaks collection-id parameter input

## Bug Description

The `CollectionSelector` component is referenced but does not exist, causing a runtime error when users interact with charts that use the `collection-id` parameter.

## Affected Files

- `apps/web/lib/charts-core/parameters/react/collection-id/index.tsx` (line 13)
- `apps/web/lib/charts-core/parameters/react/index.tsx` (parameter mapping)

## Error

```
Module not found: Can't resolve '@/components/ui/components/CollectionSelector'
```

## Impact

- Charts using `collection-id` parameter (e.g., `charts/collection-annually-ranking/`) will fail to render
- Users cannot select collections in parameter inputs
- Runtime crash when loading affected charts

## Reproduction

1. Navigate to any chart that uses `collection-id` parameter
2. The page will fail to load due to missing component

## Suggested Fix

Create a `CollectionSelector` component similar to `GHOrgSelector` or `GHRepoSelector`:

1. Create `apps/web/components/ui/components/CollectionSelector/index.ts`
2. Create `apps/web/components/ui/components/CollectionSelector/CollectionSelector.tsx`
3. Use the `/api/collections` endpoint to fetch collection list
4. Implement search and selection functionality

## Related

- Similar components: `GHOrgSelector`, `GHRepoSelector`, `RemoteSelector`
- API endpoint: `apps/web/app/api/collections/route.ts`

--- a/apps/web/components/ui/components/CollectionSelector/CollectionSelector.tsx
+++ b/apps/web/components/ui/components/CollectionSelector/CollectionSelector.tsx
@@ -1,0 +1,52 @@
+import { RemoteSelector, RemoteSelectorProps } from '../RemoteSelector';
+import { RemoteSelectItem, RemoteSelectorListItemProps } from '../RemoteSelector';
+import { RemoteSelectedItem, RemoteSelectedItemCommonProps } from '../RemoteSelector';
+import { getCollectionText, isCollectionEquals, searchCollections } from './utils';
+
+export type CollectionInfo = {
+  id: number;
+  name: string;
+};
+
+export interface CollectionSelectorProps extends Pick<RemoteSelectorProps<any>, 'id' | 'renderInput'> {
+  collection: CollectionInfo | undefined;
+  onCollectionSelected: (collection: CollectionInfo | undefined) => void;
+}
+
+export function CollectionSelector({ collection, onCollectionSelected, ...props }: CollectionSelectorProps) {
+  return (
+    <RemoteSelector<CollectionInfo>
+      {...props}
+      getItemText={getCollectionText}
+      value={collection ? [collection] : []}
+      onSelect={onCollectionSelected}
+      getRemoteOptions={searchCollections}
+      executeOnMount
+      renderSelectedItems={([item]) => (
+        <CollectionItem id={props.id} item={item} onClear={() => onCollectionSelected(undefined)} />
+      )}
+      renderListItem={itemProps => <CollectionListItem key={itemProps.item.id} {...itemProps} />}
+      equals={isCollectionEquals}
+    />
+  );
+}
+
+function CollectionItem({ item, ...props }: RemoteSelectedItemCommonProps & { item: CollectionInfo }) {
+  return (
+    <RemoteSelectedItem {...props}>
+      <span className="overflow-hidden whitespace-nowrap text-ellipsis">
+        {item.name}
+      </span>
+    </RemoteSelectedItem>
+  );
+}
+
+function CollectionListItem({ item, ...props }: RemoteSelectorListItemProps<CollectionInfo>) {
+  return (
+    <RemoteSelectItem {...props}>
+      <span className="overflow-hidden whitespace-nowrap text-ellipsis">
+        {item.name}
+      </span>
+    </RemoteSelectItem>
+  );
+}

--- a/apps/web/components/ui/components/CollectionSelector/index.ts
+++ b/apps/web/components/ui/components/CollectionSelector/index.ts
@@ -1,0 +1,1 @@
+export * from './CollectionSelector';

--- a/apps/web/components/ui/components/CollectionSelector/utils.ts
+++ b/apps/web/components/ui/components/CollectionSelector/utils.ts
@@ -1,0 +1,26 @@
+import { API_SERVER } from '@/utils/api';
+import { CollectionInfo } from './CollectionSelector';
+
+export async function searchCollections(text: string, signal?: AbortSignal): Promise<CollectionInfo[]> {
+  const params = new URLSearchParams();
+  if (text) {
+    params.set('keyword', text);
+  }
+  const response = await fetch(
+    `${API_SERVER}/collections/api?${params.toString()}`,
+    { signal },
+  );
+  if (!response.ok) {
+    throw new Error(`Failed to search collections: ${response.status}`);
+  }
+  const result: { data: Array<{ id: number; name: string }> } = await response.json();
+  return (result.data ?? []).map(({ id, name }) => ({ id, name }));
+}
+
+export function getCollectionText(collection: CollectionInfo) {
+  return collection.name;
+}
+
+export function isCollectionEquals(a: CollectionInfo, b: CollectionInfo) {
+  return a.id === b.id;
+}

